tests/test_dte_data_models.py

```diff
@@ -222,70 +298,270 @@ def test_constants_match(self):
+            self.assertIn(expected_validation_error, validation_errors)
+
+    def test_validate_receptor_razon_social_empty(self) -> None:
+        expected_validation_errors = [
+            {
+                'loc': ('receptor_razon_social',),
+                'msg': "Value must not be empty.",
+                'type': 'value_error',
+            },
+        ]
+
+        with self.assertRaises(pydantic.ValidationError) as assert_raises_cm:
             dataclasses.replace(
                 self.dte_l2_1,
                 receptor_razon_social='',
             )
-        self.assertEqual(cm.exception.args, ("Value must not be empty.", ))
 
-    def test_init_ok_razon_social_none(self) -> None:
-        _ = dataclasses.replace(
-            self.dte_l2_1,
-            emisor_razon_social=None,
-            receptor_razon_social=None,
-        )
+        validation_errors = assert_raises_cm.exception.errors()
+        self.assertEqual(len(validation_errors), len(expected_validation_errors))
+        for expected_validation_error in expected_validation_errors:
+            self.assertIn(expected_validation_error, validation_errors)
 
-    def test_init_fail_regression_signature_value_bytes_with_x20(self) -> None:
-        bytes_value_with_x20_as_base64 = 'IN2pkDBxqDnGl4Pfvboi'
-        bytes_value_with_x20 = b'\x20\xdd\xa9\x900q\xa89\xc6\x97\x83\xdf\xbd\xba"'
+    def test_validate_datetime_tz(self) -> None:
+        # Test TZ-awareness:
 
-        self.assertEqual(b'\x20', b' ')
-        self.assertEqual(
-            bytes_value_with_x20,
-            base64.b64decode(bytes_value_with_x20_as_base64, validate=True))
+        expected_validation_errors = [
+            {
+                'loc': ('firma_documento_dt',),
+                'msg': 'Value must be a timezone-aware datetime object.',
+                'type': 'value_error',
+            },
+        ]
+
+        with self.assertRaises(pydantic.ValidationError) as assert_raises_cm:
+            dataclasses.replace(
+                self.dte_l2_1,
+                firma_documento_dt=datetime(2019, 4, 5, 12, 57, 32),
+            )
 
-        init_kwars = self.dte_l2_1.as_dict()
-        init_kwars.update(dict(signature_value=bytes_value_with_x20))
+        validation_errors = assert_raises_cm.exception.errors()
+        self.assertEqual(len(validation_errors), len(expected_validation_errors))
+        for expected_validation_error in expected_validation_errors:
+            self.assertIn(expected_validation_error, validation_errors)
 
-        # with self.assertRaises(ValueError) as cm:
-        #     _ = DteDataL2(**init_kwars)
-        # self.assertEqual(
-        #     cm.exception.args,
-        #     ('Value has leading or trailing whitespace characters.', bytes_value_with_x20)
-        # )
-        _ = DteDataL2(**init_kwars)
+        # Test TZ-value:
 
-    def test_init_fail_regression_signature_cert_der_bytes_with_x20(self) -> None:
```

Comment:
Is this test method not needed anymore?

Reply:
Replaced by `DteDataL2Test.test_validate_non_empty_bytes_signature_x509_cert_der`

Reply:
See https://github.com/fyntex/lib-cl-sii-python/pull/234#discussion_r685426432.

Reply:
Restored